js/bare_board.js

```diff
@@ -327,64 +239,22 @@ export class Board {
 
   }
 
-  /**
-   * Generate footer callback.
-   * @returns {Function} footer callback
-   */
-  generateFooterCallback() {
-
-    this.titleManager.displayMessage(
-      'Experiment in progress...',
-      'progress');
-
-    const footerCallback = () => {
-      d3.select('.top-bar__detection__value').html(`${(100 * this.winningStatus.totalProbAtDets).toFixed(0)}%`);
-
-      this.titleManager.displayMessage(
-        this.winningStatus.message,
-        this.winningStatus.isWon ? 'success' : 'failure'
-      );
-
-      if (this.winningStatus.isWon) {
-
-        // TODO(migdal): make it more serious
-        this.storage.setItem(
-          `isWon ${this.level.group} ${this.level.name}`,
-           'true'
-        );
-        this.progressPearls.update();
-
-        d3.select('.top-bar__detection').classed('top-bar__detection--success', true);
-        if (this.level.group === 'Game') {
-          // TODO(pathes): make a separate component for detection % and next level button
-          d3.select('.top-bar__detection__caption').html('next level »');
-          d3.select('.top-bar__detection').on('click', () => {
-            this.logger.logAction('nextLevelButton');
-            this.loadLevel(this.level.next);
-          });
-        }
-      }
-
-      this.particleAnimation = null;
-    };
-
-    return footerCallback;
-  }
 
   /**
    * Play animation. Generate history if necessary.
    */
+  // TODO simplify its logic?
   play() {
     this.logger.logAction('simulationPlay');
-    if (!this.particleAnimation) {
+    this.callbacks.animationStart();
+    if (!this.animationExists) {
       this.generateHistory();
       this.particleAnimation = new CanvasParticleAnimation(
         this,
         this.simulationQ.history,
         this.simulationQ.measurementHistory,
         this.winningStatus.absorptionProbabilities,
-        this.generateFooterCallback());
-      this.saveProgress();
```

Comment:
Nie podoba mi się taka zmiana, że przy okazji refactoru - rozbijania boardów - znika taki feature.

Czy widzisz jakiś powód dlaczego by nie zapisywać stanu obecnej planszy w local storage'u przy każdym playu? To nie jest kosztowne, a potrafi ochronić przed utratą postępu. Zapisywanie na `onbeforeunload` nie jest niezawodne.